examples/o1/Using_chained_calls_for_o1_structured_outputs.ipynb

```diff
@@ -0,0 +1,266 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Using chained calls for reasoning structured outputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The initially released versions (September 2024) of [o1](https://openai.com/index/introducing-openai-o1-preview/) reasoning models have advanced capabilities but do not have [structured outputs](https://platform.openai.com/docs/guides/structured-outputs/examples) support. \n",
+    "\n",
+    "This means that requests with o1 don't have reliable type-safety and rely on the prompt itself to return a useful JSON. \n",
+    "\n",
+    "In this guide, we'll explore two methods to prompt o1 models, specifically `o1-preview`, to return a valid JSON format when using the OpenAI API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Prompting\n",
+    "\n",
+    "The simplest way to return a JSON response using `o1-preview` is to explicitly prompt it. \n",
+    "\n",
+    "Let's run through an example of:\n",
+    "- Fetching a wikipedia page of companies\n",
+    "- Determining which could benefit the most from AI capabilities\n",
+    "- Returning them in a JSON format, which could then be ingested by other systems"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "    \"companies\": [\n",
+      "        {\n",
```

Comment:
Double check the outputs of this cell. I can see in the markdown version that it outputs the json twice

Reply:
Oh I think I just put it in my text manually, I'll just remove it